src/colorizer/ColorizeCanvas3D.ts

```diff
@@ -430,4 +447,72 @@ export class ColorizeCanvas3D implements IRenderCanvas {
     }
     return null;
   }
+
+  public getScreenSpaceMatrix(): Matrix4 {
+    if (!this.volume) {
+      // Return an identity matrix if the volume is not loaded
+      return new Matrix4();
+    }
+
+    // 1. Normalize from volume voxel coordinates to world space. Also,
+    //    translate so that the center of the volume is at (0, 0, 0).
+    const volumeScale = new Vector3(1, 1, 1)
+      .multiply(this.volume.physicalPixelSize)
+      .divideScalar(this.volume.physicalScale);
+    const normalizeVoxelToWorld = new Matrix4().compose(
+      this.volume.normPhysicalSize.clone().multiplyScalar(-0.5), // Translate to center
+      new Quaternion(0, 0, 0, 1),
+      volumeScale
+    );
+
+    // 2. Get the view projection matrix, which transforms from world space to
+    //    screen space in the [-1, 1] range.
+    const viewProjectionMatrix = this.view3d.getViewProjectionMatrix();
+
+    // 3. Scale the [-1, 1] range to canvas pixels, and move the origin to the
+    //    top left corner of the canvas. Normalize the Z axis to the [0, 1] range.
+    const viewProjectionToScreen = new Matrix4().compose(
+      new Vector3(0.5 * this.canvasResolution.x, 0.5 * this.canvasResolution.y, 0.5), // Translate origin
+      new Quaternion(0, 0, 0, 1),
+      new Vector3(0.5 * this.canvasResolution.x, -0.5 * this.canvasResolution.y, 0.5) // Scale to screen
+    );
+
+    return viewProjectionToScreen.multiply(viewProjectionMatrix).multiply(normalizeVoxelToWorld);
+  }
+
+  public getDepthToScaleFn(screenSpaceMatrix: Matrix4): (depth: number) => { scale: number; clipOpacity: number } {
+    if (!this.volume) {
+      return () => ({ scale: 1, clipOpacity: 1 });
+    }
+    // Determine min and max Z depth of the volume in screen space, using the
+    // corners.
+    const xCoords = [0, this.volume.imageInfo.originalSize.x - 1];
+    const yCoords = [0, this.volume.imageInfo.originalSize.y - 1];
+    const zCoords = [0, this.volume.imageInfo.originalSize.z - 1];
+
+    let minZ = Infinity;
+    let maxZ = -Infinity;
+    for (let i = 0; i < xCoords.length; i++) {
+      for (let j = 0; j < yCoords.length; j++) {
+        for (let k = 0; k < zCoords.length; k++) {
+          const point = new Vector3(xCoords[i], yCoords[j], zCoords[k]);
+          point.applyMatrix4(screenSpaceMatrix);
+          minZ = Math.min(minZ, point.z);
+          maxZ = Math.max(maxZ, point.z);
+        }
+      }
+    }
```

Comment:
nit: nothing wrong here, but this could also use a three.Box3 and Box3.applyMatrix4()